Map enums using array to the correct column type (#51124)
rails/rails: activerecord/CHANGELOG.md

> *   If enum is defined without keywords, auto-detect the column type and cast the value accordingly.

Reviewer: ```suggestion
*   Auto-detect the column type for enums defined with the array syntax.
```

>     Previously enums without keywords were always mapped to integers
>     and indexed by position, now it is converted based on the column type,
>     it works for strings and database enums as well.

Reviewer: ```suggestion
    Previously enums defined using an array were always mapped to integers
    and indexed by position, now it is converted based on the column type,
    it works for strings and database enums as well.
```

    ```ruby
    # Before

>     enum status: { proposed: "proposed", written: "written", published: "published" }

Reviewer: ```suggestion
    enum :status, { proposed: "proposed", written: "written", published: "published" }
```

    # After

>     enum status: [:proposed, :written, :published]

Reviewer: ```suggestion
    enum :status, [:proposed, :written, :published]
```
